src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsTests/SettingsTests.tsx

```diff
@@ -240,65 +163,35 @@ const SettingsTests: React.FC<{openDefinition(): void}> = ({openDefinition}) =>
             Learn more about <ExternalLink href={externalLinks.testSuitesCreating}>Tests in a test suite</ExternalLink>
           </>
         }
-        onConfirm={saveSteps}
-        onCancel={() => setCurrentSteps(initialSteps)}
+        onConfirm={onSave}
+        onCancel={() => setSteps(initialSteps)}
         isButtonsDisabled={!wasTouched}
         isEditable={mayEdit}
         enabled={mayEdit}
         forceEnableButtons={wasTouched}
       >
-        <>
-          {currentSteps?.length === 0 ? (
-            <EmptyTestsContainer>
-              <Title level={2} className="text-center">
-                Add your tests to this test suite
-              </Title>
-              <Text className="regular middle text-center">
-                Select tests from the dropdown below to add them to this suite
-              </Text>
-            </EmptyTestsContainer>
-          ) : null}
-          <DragNDropList
-            value={currentSteps}
-            onChange={setCurrentSteps}
-            onDelete={deleteStep}
-            scrollRef={scrollRef}
-            ContainerComponent={StyledStepsList}
-            ItemComponent={TestSuiteStepCard}
-            disabled={!mayEdit}
-          />
-          <DelayModal
-            isDelayModalVisible={isDelayModalVisible}
-            setIsDelayModalVisible={setIsDelayModalVisible}
-            addDelay={addDelay}
-          />
-          {mayEdit ? (
-            <Select
-              placeholder="Add a test or delay"
-              showArrow
-              onChange={onSelectStep}
-              style={{width: '100%', marginTop: 15, marginBottom: 30}}
-              value={null}
-              showSearch
-              size="middle"
-            >
-              <Option value="delay">
-                <StyledOptionWrapper>
-                  <ClockCircleOutlined />
-                  <Text className="regular middle">Delay</Text>
-                </StyledOptionWrapper>
-              </Option>
-              {allTestsData.map(item => (
-                <Option value={JSON.stringify(item)} key={item.name}>
-                  <StyledOptionWrapper>
-                    <ExecutorIcon type={item.type} />
-                    <Text className="regular middle">{item.name}</Text>
-                  </StyledOptionWrapper>
-                </Option>
-              ))}
-            </Select>
-          ) : null}
-        </>
+        {steps?.length === 0 ? (
```

Comment:
Actually, `steps?.length === 0` will be falsy when `steps === undefined`. To handle that:
```suggestion
        {!steps?.length ? (
```